core/store/ChainStore/ChainStore.go

```diff
@@ -800,6 +805,11 @@ func (bd *ChainStore) persist(b *Block) error {
 			if accounts[programHash].Balances[assetId] < 0 {
 				return errors.New(fmt.Sprintf("account programHash:%v, assetId:%v insufficient of balance", programHash, assetId))
 			}
+
+			err = bd.modifyUnspentWithProgramHash(programHash, assetId, transaction.Hash(), index, -1)
```

Comment:
One write op one block. It's inefficient for every input and output to operator db.